### PR TITLE
Miscellaneous info

### PR DESCRIPTION
**Pull request by Zorrow2008**
This PR contains a dynamic graph which plots number of nonzero words over time.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -30,6 +30,7 @@
   },
   "homepage": "",
   "devDependencies": {
+    "@babel/plugin-proposal-class-properties": "^7.8.3",
```
Comment (swkeever) [inline package.json]: what's this?

File: src/components/MemoryGraph.js [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+  console.log(numArray);
+  const data = {
+
+    // i want to come up with better label names
+    // these are also random style attributes which i copied, but the color does look nice
+    labels: ['10', '9', '8', '7', '6', '5', '4', '3', '2', '1'],
+    datasets: [
+      {
+        label: 'My First dataset',
```
Comment (swkeever) [inline src/components/MemoryGraph.js]: we would want to style this in CSS and not directly in react but feel free to keep it for now

File: src/components/MiscellaneousPanel.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import React from 'react';
+import useCPU from '../hooks/use-cpu';
+import Loader from './Loader';
+import MemoryGraph from './MemoryGraph';
+
+// Honestly, we might think about getting rid of this component
+// its just a placeholder to store the graph, and potentially endian feature
+const MiscellaneousPanel = () => {
```
Comment (swkeever) [inline src/components/MiscellaneousPanel.js]: why is it called miscellaneous panel? i thought it was for showing memory utilization?

File: public/assets/webworker.js [hunk 4]
```diff
@@ -71,16 +71,27 @@ function sendCpuState() {
...
 
+  // compute num of nonzero words in memory
+  var num = 0;
+  for (const [index, element] of RISCV.memory.entries()) {
+    if (element != 0) {
+      num++;
+    }
+  }
+  // console.log("total ", RISCV.memory.length);
+  // console.log("num: ", num);
```
Comment (swkeever) [inline public/assets/webworker.js]: can we remove commented out debug code?

File: public/index.html [hunk 5]
```diff
@@ -37,6 +37,7 @@
   <!-- <div id="react"></div> -->
   <div id="registerPanel"></div>
   <div id="instructionDisplay"></div>
+  <div id="miscellaneousPanel"></div>
```
Comment (swkeever) [inline public/index.html]: can we change this from misc panel to like memoryPanel or something?

also can we change instructionDisplay to instructionPanel so its more consistent? (if you change instructionDisplay make sure to also change the selector in public/assets/styles/style.css)

File: src/components/MemoryGraph.js [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+      const temp = numArray.filter(((ele, i) => i !== 0));
+      const newArr = temp.concat(next);
+      setRange(newArr);
+      addToNumArray(newArr);
+    } else {
+      const newArr = numArray.concat(next);
+      setRange(newArr);
+      addToNumArray(newArr);
+    }
+  }, 1000);
```
Comment (swkeever) [inline src/components/MemoryGraph.js]: should we define 1000 as a constant in our constants file? i'm worried we are setting this 1000 number in a lot of places

File: src/components/MemoryGraph.js [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+
+  console.log(numArray);
+  const data = {
+
+    // i want to come up with better label names
+    // these are also random style attributes which i copied, but the color does look nice
+    labels: ['10', '9', '8', '7', '6', '5', '4', '3', '2', '1'],
```
Comment (swkeever) [inline src/components/MemoryGraph.js]: what do you think the label names should be? should we figure these out now?

File: src/components/MemoryGraph.js [hunk 8]
```diff
@@ -0,0 +1,106 @@
...
+import PropTypes from 'prop-types';
+import { Line } from 'react-chartjs-2';
+import useInterval from '../hooks/use-interval';
+
+/* this line chart graphs the last 10 data points we've seen,
+and updates the lower and upper bound of the range to be equal
+to the min and max of this dataset, respectively. */
+const numData = 10;
+
+const MemoryGraph = ({ next }) => {
```
Comment (swkeever) [inline src/components/MemoryGraph.js]: what is next supposed to be?

File: src/components/MemoryGraph.js [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+  const [numArray, addToNumArray] = useState([]);
+  const [min, setMin] = useState(0);
+  const [max, setMax] = useState(2500000);
```
Comment (swkeever) [inline src/components/MemoryGraph.js]: what is 2500000? should that be a constant variable with a descriptive name?

File: src/components/MemoryGraph.js [hunk 10]
```diff
@@ -0,0 +1,106 @@
...
+
+  const setRange = (newArr) => {
+    setMin(Math.min(...newArr));
+    setMax(Math.max(...newArr));
+  };
+
+  useInterval(() => {
+    // const answer = (parseFloat(next) / total) * 100
```
Comment (swkeever) [inline src/components/MemoryGraph.js]: do we need to use useInterval here? the parent element is using useCpu which uses useInterval, and you are getting state from the parent. i'm not sure you need useInterval

File: package.json (hunk 1 above)
Comment (Zorrow2008) [reply package.json]: its from when i was playing around with dependencies; will remove

File: src/components/MiscellaneousPanel.js (hunk 3 above)
Comment (Zorrow2008) [reply src/components/MiscellaneousPanel.js]: im going to remove this component and just put the memory component instead

File: public/assets/webworker.js (hunk 4 above)
Comment (Zorrow2008) [reply public/assets/webworker.js]: done

File: src/components/MemoryGraph.js (hunk 6 above)
Comment (Zorrow2008) [reply src/components/MemoryGraph.js]: sure, ill add this

File: src/components/MemoryGraph.js (hunk 7 above)
Comment (Zorrow2008) [reply src/components/MemoryGraph.js]: i think its ok like this, theyre just enumerating the last X datapoint we've seen. maybe when youre playing with the graph, you can see if theres something in the library that allows you to add labels to axis, which im sure there is

File: src/components/MemoryGraph.js (hunk 8 above)
Comment (Zorrow2008) [reply src/components/MemoryGraph.js]: i removed the miscellaneousDisplay component, and am now calling MemoryDisplay (used to be MemoryGraph), and removed this prop

File: src/components/MemoryGraph.js (hunk 9 above)
Comment (Zorrow2008) [reply src/components/MemoryGraph.js]: ill add this

File: src/components/MemoryGraph.js (hunk 10 above)
Comment (Zorrow2008) [reply src/components/MemoryGraph.js]: am now using useEffect

File: public/index.html (hunk 5 above)
Comment (Zorrow2008) [reply public/index.html]: renamed everything to end with Display

Comment (Zorrow2008) [reply public/index.html]: oh, Ill change everything to end with Panel, actually.

**Pull request merged**